Microsoft.SystemForCrossDomainIdentityManagement/Protocol/PatchOperation2Combined.cs

```diff
@@ -41,6 +42,37 @@ public string Value
             }
         }
 
+        [OnDeserialized]
+        private void OnDeserialized(StreamingContext context)
+        {
+            this.OnInitialized();
+        }
+
+        [OnDeserializing]
+        private void OnDeserializing(StreamingContext context)
+        {
+            this.OnInitialization();
+        }
+
+        private void OnInitialization()
+        {
+        }
+
+        private void OnInitialized()
+        {
+            if (this.values == null && this.Path == null)
+            {
+                throw new NotSupportedException(SystemForCrossDomainIdentityManagementProtocolResources.ExceptionInvalidValue);
+            }
+            else if(this.values == null)
+            { 
+                if (this.Path.SubAttributes.Any())
+                {
+                    this.values = this.Path.SubAttributes.First().ComparisonValue;
+                }
```

Comment:
None of this should be required if the deserailization of the request is happening correctly.